app/src/main/java/org/mozilla/fenix/settings/creditcards/view/CreditCardsManagementView.kt

```diff
@@ -41,6 +41,9 @@ class CreditCardsManagementView(
         credit_cards_list.isVisible = state.creditCards.isNotEmpty()
 
         creditCardsAdapter.submitList(state.creditCards)
+        if (!state.isLoading && state.creditCards.isEmpty()) {
```

Comment:
I am trying to stick with @Mugurell's philosophy around `View` described here https://github.com/mozilla-mobile/fenix/pull/18808#discussion_r608822058. I am wondering if we should just do this check at https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/settings/creditcards/CreditCardsManagementFragment.kt#64 to avoid having this logic in the View.

Otherwise, I would recommend moving this to top of `update` to avoid the unnecessary UI updates before popping the nav stack.

@Mugurell also interested in your thoughts here.

Reply:
Perhaps another approach would be to check if we have any credit cards after deleting and decide where to navigate in https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/settings/creditcards/controller/CreditCardEditorController.kt#65.

Reply:
This is an interesting discussion.
We have Middlewares that sit between Actions and Reducers but no clear way to intercept and act on new States before they get to Views.
Asked Christian about this and he recommended "Handling the state change in consumeFrom is the right place. You can decide right there in the fragment that the view should be closed and call a method on the view directly."

Reply:
@gabrielluong Following the first suggestion I would rather have `CreditCardsManagementFragment` decide to navigate back before updating the view.

Reply:
Throughout the app there are various ways to handle this situation. It would be great to focus on one particular approach from now on.